src/test/java/org/jenkinsci/plugins/workflow/properties/PipelineTriggersJobPropertyTest.java

```diff
@@ -0,0 +1,177 @@
+        assertEquals("[false]", t.calls.toString());
+        assertTrue(t.isStarted);
+    }
+
+    @Test
+    public void submitCallsStartTrue() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        p = (WorkflowJob) r.configRoundtrip((Item)p);
+        t = (MockTrigger)p.getTriggers().get(t.getDescriptor());
+        assertNotNull(t);
+        assertEquals("[true]", t.calls.toString());
+        assertTrue(t.isStarted);
+    }
+
+    @Test
+    public void previousTriggerStopped() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        assertTrue(t.isStarted);
+        MockTrigger t2 = new MockTrigger();
+        p.addTrigger(t2);
+        p.save();
+
+        assertFalse(t.isStarted);
+        assertTrue(t2.isStarted);
+    }
+
+    @LocalData
+    @Test
+    public void triggerMigration() throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+        assertNotNull(p);
+
+        PipelineTriggersJobProperty triggerProp = p.getProperty(PipelineTriggersJobProperty.class);
+        assertNotNull(triggerProp);
+        assertEquals(1, triggerProp.getTriggers().size());
+        assertEquals(1, p.getTriggers().size());
+
+        Trigger fromProp = triggerProp.getTriggers().get(0);
+        assertEquals(TimerTrigger.class, fromProp.getClass());
+
+        Trigger fromJob = p.getTriggers().get(fromProp.getDescriptor());
+        assertEquals(fromProp, fromJob);
+    }
+
+    @Test
+    public void configRoundTrip() throws Exception {
+        WorkflowJob defaultCase = r.jenkins.createProject(WorkflowJob.class, "defaultCase");
+        assertTrue(defaultCase.getTriggers().isEmpty());
+
+        WorkflowJob roundTripDefault = r.configRoundtrip(defaultCase);
+        assertTrue(roundTripDefault.getTriggers().isEmpty());
+
+        WorkflowJob withTriggerCase = r.jenkins.createProject(WorkflowJob.class, "withTriggerCase");
+        withTriggerCase.addTrigger(new MockTrigger());
+        assertEquals(1, withTriggerCase.getTriggers().size());
+        List<Trigger<?>> origTriggers = new ArrayList<>(withTriggerCase.getTriggers().values());
+
+        assertEquals(MockTrigger.class, origTriggers.get(0).getClass());
+
+        WorkflowJob roundTripWithTrigger = r.configRoundtrip(withTriggerCase);
+
+        assertEquals(1, roundTripWithTrigger.getTriggers().size());
+        List<Trigger<?>> modTriggers = new ArrayList<>(roundTripWithTrigger.getTriggers().values());
+
+        assertEquals(MockTrigger.class, modTriggers.get(0).getClass());
+    }
+
+    public static class MockTrigger extends Trigger<Item> {
+
+        public transient List<Boolean> calls = new ArrayList<Boolean>();
+        public transient boolean isStarted = false;
```

Comment:
Might be more convincing for these to be replaced with a single

```java
/** elements are true or false for {@code start}, null for {@code stop} */
public static List<Boolean> startsAndStops = new ArrayList<>();
```

so that you can demonstrate that the correct number and kind of calls are made on both old and new instances: that there are not, for example, temporary `Trigger`s constructed, `start`ed, then discarded without being `stop`ped.

This would be closer to the real behavior of `Trigger`s overriding these methods, which tend to do things like register webhooks etc. with some global effect, where you want to ensure that calls are properly paired.